stl/inc/xstring

```diff
@@ -2932,7 +2950,7 @@ public:
     struct _Released_buffer {
         pointer _Ptr;
         size_type _Size;
-        size_type _Res;
+        size_type _Res; // real allocation size

```

Comment:
I think `_Res` is a terrible name as it is conflicting with that of `_String_val::_Myres`. In what condition can we change the name of a data member?

Reply:
Messing with *types* poses grave ABI concerns. For ABI-locked code (i.e. older than `/std:c++latest`), we can't ever do this for data members of types that the user can directly create or derive from.

We can mess with types when they are purely internal - e.g. a struct that is used only as a local variable within a function. In such cases, we should rename the type to avoid ODR issues.

In certain cases we can also rename internal types when their manipulation is purely through virtuals; this happened with `shared_ptr`'s control blocks, for example.

Messing with names will break visualizers (if present) and we shouldn't do it without a very good reason.

With `_Released_buffer::_Res`, maybe we can rename it.

Reply:
Renamed to `_Actual_allocation_size`.

Reply:
I've checked the changes to `mutex` in pr #3824. It looks I can move `monostate` back to `<variant>`  by dropping reliance from `basic_format_arg`.
I think defining `monostate` in `<xutility>` is undesirable, as the standard require it to be defined in `<variant>`, and all supporting utilities (compare/hash) and use cases remain in `<variant>`.
This is the change: https://github.com/microsoft/STL/compare/main...achabense:STL:_Cleanups, could you help me check whether it is a safe approach? (There is another place I'm not confident about in `<exception>` in the branch)

Reply:
That seems reasonable, although I would simply direct-init `_Monostate _No_state{};`.

Changing `_With_nested`'s template parameters should affect mangling, making it a safe change (internal class that users can't directly store). I think it would be better to rename it with `_v2` though, just to be extra safe.